examples/exec-healthz/simpleHealthzServer.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// A tiny web server that returns 200 on it's healthz endpoint if the command
+// passed in via -cmd exits with 0. Returns 503 otherwise.
+// Usage: simpleHealthzServer -port 8080 -cmd 'nslookup localhost >/dev/null'
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+// TODO:
+// 1. Sigterm handler for docker stop
+// 2. Meaningful default healthz
+// 3. Timeout exec.Command
+// 4. 404 for unknown endpoints
+
+var (
+	port = flag.Int("port", 8080, "Port number to serve /healthz.")
+	cmd  = flag.String("cmd", "echo healthz", "Command to run in response to a GET on /healthz. If the given command exits with 0, /healthz will respond with a 200.")
+)
+
+func main() {
+	flag.Parse()
+	links := []struct {
+		link, desc string
+	}{
+		{"/healthz", "healthz probe. Returns \"ok\" if the command given through -cmd exits with 0."},
+		{"/quit", "Cause this container to exit."},
+	}
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "<b> Kubernetes healthz sidecar container </b><br/><br/>")
+		for _, v := range links {
+			fmt.Fprintf(w, `<a href="%v">%v: %v</a><br/>`, v.link, v.link, v.desc)
+		}
+	})
+
+	http.HandleFunc("/quit", func(w http.ResponseWriter, r *http.Request) {
+		os.Exit(0)
+	})
+	http.HandleFunc("/healthz", healthzHandler)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", *port), nil))
+}
+
+func healthzHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I really think you need to rate-limit this or just run it async and return the most recent result.  It's too ripe for accidental DoS